server/odc-service/src/main/java/com/oceanbase/odc/service/task/caller/ExecutorProcessBuilderFactory.java

```diff
@@ -31,35 +34,46 @@
  */
 public class ExecutorProcessBuilderFactory {
 
+    private static final Pattern ODC_SERVER_EXECUTABLE_JAR = Pattern.compile("^.*odc-server-.*executable\\.jar$");
+
     public ProcessBuilder getProcessBuilder(Map<String, String> environments, String executorName) {
         RuntimeMXBean runtimeMxBean = ManagementFactory.getRuntimeMXBean();
         ProcessBuilder pb = new ProcessBuilder();
         List<String> commands = new ArrayList<>();
         commands.add("java");
-        commands.add("-D" + JobConstants.ODC_EXECUTOR_PROCESS_PROPERTIES_KEY + "=" + executorName);
-        commands.addAll(jvmOptions());
-        commands.add("-classpath");
-        commands.add(runtimeMxBean.getClassPath());
-        commands.add(JobConstants.ODC_SERVER_CLASS_NAME);
-        pb.directory(new File("."));
+        commands.add("-D" + JobUtils.generateExecutorSelectorOnProcess(executorName));
+        commands.addAll(jvmOptions(environments));
+        if (ODC_SERVER_EXECUTABLE_JAR.matcher(runtimeMxBean.getClassPath()).matches()) {
+            // start odc executor by java -jar
+            commands.add("-jar");
+            // set jar package file name in commands
+            commands.add(runtimeMxBean.getClassPath());
+        } else {
+            // start odc executor by java -classpath
+            commands.add("-classpath");
+            commands.add(runtimeMxBean.getClassPath());
+            commands.add(JobConstants.ODC_SERVER_CLASS_NAME);
+        }
         pb.command(commands);
+        pb.directory(new File("."));
         pb.environment().putAll(environments);
         return pb;
     }
 
-    private List<String> jvmOptions() {
+    private List<String> jvmOptions(Map<String, String> environments) {
         List<String> options = new ArrayList<>();
         options.add("-XX:+UseG1GC");
         options.add("-XX:+PrintAdaptiveSizePolicy");
         options.add("-XX:+PrintGCDetails");
         options.add("-XX:+PrintGCTimeStamps");
         options.add("-XX:+PrintGCDateStamps");
+        options.add(String.format("-Xloggc:%s/gc.log", environments.get(JobEnvKeyConstants.ODC_LOG_DIRECTORY)));
         options.add("-XX:+UseGCLogFileRotation");
         options.add("-XX:GCLogFileSize=50M");
         options.add("-XX:NumberOfGCLogFiles=5");
         options.add("-XX:+ExitOnOutOfMemoryError");
         options.add("-Xmx2048m");
-        options.add("-Xms2048m");
+        options.add("-Xms1024m");
```

Comment:
avoid hard code heap size, what if a task require more memory in prod env ?

Reply:
ok, task can set jvm memory by  `SingleJobProperties`  when  submit JobDefinition to  task-framework, this will modify in later feature pr。